### DISCUSSION TITLE
Get device address from notification callback

### DISCUSSION THREAD

**Discussion by TheAlbertDev** [hbldh/bleak#1188] [Q&A] [answered]
I receive data correctly in the callback that is executed when a characteristic is updated. This characteristic is sent to me as a parameter to the callback. How can I find out which device has made the communication (I have several connected)? I would like to get the address of the device.

Best regards and thanks for your time.

**Comment by dlech** [comment]
There is a feature request for this in https://github.com/hbldh/bleak/issues/888. In the mean time, you can use closures, lambda expressions, `functools.partial`, etc. to capture the device address when the callback function is created.

**Reply by TheAlbertDev** [reply]
I'll take a look 😉
Thx!

**Comment by TheAlbertDev** [comment, marked as answer]
Implementated answer here #467.
